src/vivarium/framework/values/pipeline.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any, TypeVar
+
+import pandas as pd
+
+from vivarium.framework.resource import Resource
+from vivarium.framework.values.exceptions import DynamicValueError
+
+if TYPE_CHECKING:
+    from vivarium.framework.values.combiners import ValueCombiner
+    from vivarium.framework.values.manager import ValuesManager
+    from vivarium.framework.values.post_processors import PostProcessor
+
+T = TypeVar("T")
```

Comment:
What is this for?

Reply:
Take a look at `_get_property()` below. It has an input type of `T | None` and a return type of `T`. This means that can take any type or None as an input and return exactly the same type that was input. i.e. if type `Foo` was passed in, you're guaranteed to get `Foo` out.